File: src/test/resources/unittests/regression/nested_constructors/nested_constructor.cpp
#include <string>

/*
    Pseudo Botan classes to simplify the example.
*/
namespace Botan {
    class AlgorithmIdentifier
    {
    };

    class RSA_PublicKey {
        public:
        PubKey(AlgorithmIdentifier id, int size)
        {
        }
    };

    class PK_Verifier
    {
        public:
        PK_Verifier(RSA_PublicKey ok, std::string algo)
        {
       }
    };
}

/* Example of nested constructors, aka temporaries in C++ */
int main()
{
    Botan::AlgorithmIdentifier id;
    Botan::PK_Verifier sig_verifier(Botan::RSA_PublicKey(id, 123), "I AM INCORRECT EMSA4(SHA-256)");
}

File: src/test/resources/unittests/regression/nested_constructors/NestedConstructor.java
package botan;

public class NestedConstructor {

	/* Example of nested constructors */
	public static void main(String... args) {
		AlgorithmIdentifier id = new AlgorithmIdentifier();
		PK_Verifier sig_verifier = new PK_Verifier(new RSA_PublicKey(id, 123), "I AM INCORRECT EMSA4(SHA-256)");
	}
}

class AlgorithmIdentifier {}

class RSA_PublicKey {
	public RSA_PublicKey(AlgorithmIdentifier id, int length) {}
}

class PK_Verifier{
	public PK_Verifier(RSA_PublicKey pk, String algorithm) {

	}
}